### ISSUE TITLE
Bocfel assertion failure on sorting

### ISSUE THREAD

**Issue by angstsmurf** [garglk/garglk#896]
Since upgrading to Xcode 16.0, Bocfel will assert at startup with `_LIBCPP_ASSERT_SEMANTIC_REQUIREMENT`: "Your comparator is not a valid strict-weak ordering" at line 473 in `options.cpp`:

https://github.com/garglk/garglk/blob/da1ab09be9edfb467ca12f39ae131c13e1e97412/terps/bocfel/options.cpp#L472-L480

My guess is that this is an actual problem that a stricter LLVM catches, but it is beyond my (non-existent) C++ skills. It seems similar to this:
https://github.com/duckdb/duckdb/issues/12666

**Comment by cspiegel** [comment]
Your guess is completely correct!  I'm able to reproduce by using libc++ and setting _LIBCPP_HARDENING_MODE to _LIBCPP_HARDENING_MODE_DEBUG.

The fix is to replace the code in the lambda with this:

```c++
        unsigned char a = a_.first, b = b_.first;

        if (std::tolower(a) == std::tolower(b)) {
            return std::islower(a) && std::isupper(b);
        }

        return std::tolower(a) < std::tolower(b);
```
I've got a few other small changes so I'll push out a new Bocfel, but for the time being you can update Spatterlight with this and I'll do the same here.

**Comment by angstsmurf** [comment]
Thanks a lot. It works!

**Comment by cspiegel** [comment]
Fixed in c1ede26d2c6f48f2583fab9eb8f2d7ec9a0443ab.
